x/posts/internal/types/post.go

```diff
@@ -260,6 +260,26 @@ func (p Post) ContentsEquals(other Post) bool {
 		ArePollDataEquals(p.PollData, other.PollData)
 }
 
+// GetPostHashtags returns all the post's hashtags without duplicates
+func (p Post) GetPostHashtags() []string {
+	hashtags := HashtagRegEx.FindAllString(p.Message, -1)
+	return UniqueHashtags(hashtags)
+}
+
+// UniqueHashtags returns the given input slice without any duplicated hashtag value inside it
+func UniqueHashtags(input []string) []string {
+	unique := make([]string, 0, len(input))
+	m := make(map[string]bool)
+
+	for _, val := range input {
+		if _, ok := m[val]; !ok {
+			m[val] = true
+			unique = append(unique, strings.Trim(val, "#"))
```

Comment:
It's probably better to use `TrimLeft`: 

```suggestion
			unique = append(unique, strings.TrimLeft(val, "#"))
```